### PR TITLE
VFB-200: Add subscription channel to collections centre table

### PR DESCRIPTION
**Pull request by minalui**
## What's changed
- Collection centres table update automatically after adding a new centre
- Success message is displayed

## Screenshots / Videos
| Before     | After      |
|------------|------------|
| image | image |

## Checklist
- [x] The ticket is up-to-date - Please document any deviations from the original approach if there is any.
- [x] I have documented the testing steps for QA
- [x] I have self-reviewed this PR
- [x] Make sure you've verified it works via `npm run dev`
- [x] Make sure you've verified it works via `npm run build` and `npm run start`
- [x] Make sure you've fixed all linting problems with `npm run lint_fix`
- [x] Make sure you've tested via `npm run test:e2e`

If you have made any changes to the database... N/A
  - [ ] The migration files are up-to-date with my final set up (`npx supabase db diff -f <name_of_migration>` should create nothing at this point)
  - [ ] I have updated the typescript definitions for the database with `db:local:generate_types`
  - [ ] I have modified the seed in `supabase/seed/seed.mts` if appropriate
  - [ ] If I have modified the seed, I have also generated the seed with `npm run db:generate_seed` 
  - [ ] With my final set up, I can run `npm run dev:reset_supabase` without any errors.
  - Does this require resetting the deployed database? - YES / NO (remove as appropriate)
  - [ ] I have checked that migration can happen successfully without resetting the database, doing the following.
    - Make sure you have rebased your branch onto dev or merged dev into your branch.
    - Checkout the dev branch
    - Run `npm run post_checkout` to reset the database, including the seed data.
    - Checkout your branch
    - Run `npx supabase migrations up --include-all --local` to run all outstanding migration files
    - Check that the resulting database is what you expect, bar any seed data changes.

### CODE REVIEW

**Comment by rickaw** [comment]
As discussed, we're parking [VFB-46](https://softwiretech.atlassian.net/browse/VFB-46).

File: src/app/admin/collectionCentresTable/CollectionCentresTable.tsx [hunk 1]
```diff
@@ -41,49 +45,73 @@ const filters: Filter<CollectionCentresTableRow, string>[] = [
...
-                    return (
-                        filter.methodConfig.paginationType === PaginationType.Client &&
-                        filter.methodConfig.method(row, filter.state, filter.key)
-                    );
-                });
-            })
-        );
-    }, [primaryFilters, props.collectionCentreData]);
+        void fetchAndDisplayCollectionCentres();
+        // This requires that the DB clients, collection_centres, and families tables have Realtime turned on
```
Comment (rickaw) [inline src/app/admin/collectionCentresTable/CollectionCentresTable.tsx:74]:
```suggestion
```

File: src/app/admin/collectionCentresTable/CollectionCentresTable.tsx [hunk 2]
```diff
@@ -41,49 +45,73 @@ const filters: Filter<CollectionCentresTableRow, string>[] = [
...
-                return primaryFilters.every((filter) => {
-                    return (
-                        filter.methodConfig.paginationType === PaginationType.Client &&
-                        filter.methodConfig.method(row, filter.state, filter.key)
-                    );
-                });
-            })
-        );
-    }, [primaryFilters, props.collectionCentreData]);
+        void fetchAndDisplayCollectionCentres();
```
Comment (rickaw) [inline src/app/admin/collectionCentresTable/CollectionCentresTable.tsx:73]:
```suggestion
```

File: src/app/admin/collectionCentresTable/CollectionCentresTable.tsx [hunk 3]
```diff
@@ -41,49 +45,73 @@ const filters: Filter<CollectionCentresTableRow, string>[] = [
...
+            .subscribe((status, err) => {
+                if (subscriptionStatusRequiresErrorMessage(status, err, "collection_centres")) {
+                    setErrorMessage("Error fetching data, please reload");
+                }
+            });
+
+        return () => {
+            void supabase.removeChannel(subscriptionChannel);
+        };
+    }, [fetchAndDisplayCollectionCentres, primaryFilters]);
```
Comment (rickaw) [inline src/app/admin/collectionCentresTable/CollectionCentresTable.tsx:91]: I think this useeffect is just subscribing to db changes right? Why does this need to run when filters change?
```suggestion
    }, [fetchAndDisplayCollectionCentres]);
```

File: src/app/admin/createCollectionCentre/CreateCollectionCentreForm.tsx [hunk 4]
```diff
@@ -102,8 +104,10 @@ const CreateCollectionCentreForm: React.FC<{}> = () => {
                     />
                 ))}
 
-                {refreshRequired ? (
-                    <RefreshPageButton />
+                {addCollectionCentreIsSuccess ? (
+                    <Alert severity="success" action={<RefreshPageButton />}>
+                        Collection centre <b>{fields.name}</b> added successfully.
```
Comment (rickaw) [inline src/app/admin/createCollectionCentre/CreateCollectionCentreForm.tsx:109]: Does this mean if we successfully add a collection centre and then change the value of a field, does that changes the success message too?

**Review by rickaw — CHANGES_REQUESTED** [review]

File: src/app/admin/createCollectionCentre/CreateCollectionCentreForm.tsx (hunk 4 above)
Comment (minalui) [reply src/app/admin/createCollectionCentre/CreateCollectionCentreForm.tsx]: Good spot 🙈

File: src/app/admin/createCollectionCentre/CreateCollectionCentreForm.tsx [hunk 5]
```diff
@@ -48,6 +48,7 @@ const CreateCollectionCentreForm: React.FC<{}> = () => {
     const [submitDisabled, setSubmitDisabled] = useState(false);
 
     const [addCollectionCentreIsSuccess, setAddCollectionCentreIsSuccess] = useState(false);
+    const [newCollectionCentre, setNewCollectionCentre] = useState<string | null>(null);
```
Comment (rickaw) [inline src/app/admin/createCollectionCentre/CreateCollectionCentreForm.tsx:51]: I think the presence of `newCollectionCentre` now indicates whether adding the collection centre was successful right? I don't think we need `addCollectionCentreIsSuccess` anymore.

Comment (minalui) [reply src/app/admin/createCollectionCentre/CreateCollectionCentreForm.tsx:51]: Good point

**Review by rickaw — APPROVED** [review]

**Pull request merged**